prometheus/config.go

```diff
@@ -122,6 +149,22 @@ func (c Configuration) NewReporter(
 		opts.DefaultSummaryObjectives = values
 	}
 
+	registryCfg := c.RegistryConfig()
+	if !registryCfg.Default {
+		registerer := prom.NewRegistry()
+		if registryCfg.GoCollector {
+			if err := registerer.Register(prom.NewGoCollector()); err != nil {
+				return nil, err
+			}
+		}
+		if registryCfg.ProcessCollector {
+			if err := registerer.Register(prom.NewProcessCollector(os.Getpid(), "")); err != nil {
+				return nil, err
+			}
+		}
+		opts.Registerer = registerer
+	}
+
```

Comment:
Also need to allow opening a handler against the passed in registry.. `reporter.HTTPHandler()` only registers it for the default registry, I think there is a `reporter.HandlerFor(...)` method..

Reply:
Good call, fixed with https://github.com/uber-go/tally/pull/80 and the changes I just made to that code now to default to that registerer.